src/Altinn.App.Core/Helpers/MultiDecisionHelper.cs

```diff
@@ -0,0 +1,186 @@
+
+        foreach (string actionType in actionTypes)
+        {
+            XacmlJsonCategory actionCategory;
+            actionCategory = DecisionHelper.CreateActionCategory(actionType, true);
+            actionCategory.Id = ActionId + counter.ToString();
+            actionCategories.Add(actionCategory);
+            counter++;
+        }
+
+        return actionCategories;
+    }
+
+    private static List<XacmlJsonCategory> CreateMultipleResourceCategory(Instance instance)
+    {
+        List<XacmlJsonCategory> resourcesCategories = new();
+        int counter = 1;
+        XacmlJsonCategory resourceCategory = new() { Attribute = new List<XacmlJsonAttribute>() };
+
+        var instanceProps = GetInstanceProperties(instance);
+
+        if (instanceProps.Task != null)
+        {
+            resourceCategory.Attribute.Add(DecisionHelper.CreateXacmlJsonAttribute(XacmlResourceTaskId, instanceProps.Task, DefaultType, DefaultIssuer));
+        }
+        else if (instance.Process?.EndEvent != null)
+        {
+            resourceCategory.Attribute.Add(DecisionHelper.CreateXacmlJsonAttribute(XacmlResourceEndId, instance.Process.EndEvent, DefaultType, DefaultIssuer));
+        }
+
+        if (!string.IsNullOrWhiteSpace(instanceProps.InstanceId))
+        {
+            resourceCategory.Attribute.Add(DecisionHelper.CreateXacmlJsonAttribute(AltinnXacmlUrns.InstanceId, instanceProps.InstanceId, DefaultType, DefaultIssuer, true));
+        }
+        else if (!string.IsNullOrEmpty(instanceProps.InstanceGuid))
+        {
+            resourceCategory.Attribute.Add(DecisionHelper.CreateXacmlJsonAttribute(AltinnXacmlUrns.InstanceId, instanceProps.InstanceOwnerPartyId + "/" + instanceProps.InstanceGuid, DefaultType, DefaultIssuer, true));
+        }
+
+        resourceCategory.Attribute.Add(DecisionHelper.CreateXacmlJsonAttribute(AltinnXacmlUrns.PartyId, instanceProps.InstanceOwnerPartyId, DefaultType, DefaultIssuer));
+        resourceCategory.Attribute.Add(DecisionHelper.CreateXacmlJsonAttribute(AltinnXacmlUrns.OrgId, instanceProps.Org, DefaultType, DefaultIssuer));
+        resourceCategory.Attribute.Add(DecisionHelper.CreateXacmlJsonAttribute(AltinnXacmlUrns.AppId, instanceProps.App, DefaultType, DefaultIssuer));
+        resourceCategory.Id = ResourceId + counter;
+        resourcesCategories.Add(resourceCategory);
+
+        return resourcesCategories;
+    }
+
+    private static (string? InstanceId, string InstanceGuid, string? Task, string InstanceOwnerPartyId, string Org, string App) GetInstanceProperties(Instance instance)
+    {
+        string? instanceId = instance.Id.Contains('/') ? instance.Id : null;
+        string instanceGuid = instance.Id.Contains('/') ? instance.Id.Split("/")[1] : instance.Id;
+        string? task = instance.Process?.CurrentTask?.ElementId;
+        string instanceOwnerPartyId = instance.InstanceOwner.PartyId;
+        string org = instance.Org;
+        string app = instance.AppId.Split("/")[1];
```

Comment:
Use AppIdentifier and avoid leaking the split logic and possibly return the AppIdentifier instead of Org and App??

Reply:
Thanks! :+1: